sdk/src/types/block/slot/index.rs

```diff
@@ -62,6 +62,56 @@ impl SlotIndex {
     }
 }
 
+impl PartialEq<u64> for SlotIndex {
+    fn eq(&self, other: &u64) -> bool {
+        self.0 == *other
+    }
+}
+
+impl core::ops::Add<u64> for SlotIndex {
+    type Output = Self;
+
+    fn add(self, other: u64) -> Self {
+        Self(self.0 + other)
+    }
+}
+
+impl core::ops::Add for SlotIndex {
+    type Output = Self;
+
+    fn add(self, other: SlotIndex) -> Self {
+        Self(self.0 + other.0)
+    }
+}
+
+impl core::ops::AddAssign<u64> for SlotIndex {
+    fn add_assign(&mut self, other: u64) {
+        self.0 += other;
+    }
+}
+
+impl core::ops::Sub<u64> for SlotIndex {
+    type Output = Self;
+
+    fn sub(self, other: u64) -> Self {
+        Self(self.0 - other)
+    }
+}
+
+impl core::ops::Sub for SlotIndex {
+    type Output = Self;
+
+    fn sub(self, other: SlotIndex) -> Self {
+        Self(self.0 - other.0)
+    }
+}
+
+impl core::ops::SubAssign<u64> for SlotIndex {
+    fn sub_assign(&mut self, other: u64) {
+        self.0 -= other;
+    }
+}
```

Comment:
those can probably be derived using the `derive_more` crate.

Reply:
Not all of them no, but I can do some, sure